sdk/maps/Azure.Maps.Search/src/MapsSearchClient.cs

```diff
@@ -0,0 +1,3239 @@
+        /// <param name="credential"> A credential used to authenticate to an Azure Maps Search Service. </param>
+        /// <param name="endpoint"> server parameter. </param>
+        /// <param name="clientId"> Specifies which account is intended for usage in conjunction with the Azure AD security model.  It represents a unique ID for the Azure Maps account and can be retrieved from the Azure Maps management  plane Account API. To use Azure AD security in Azure Maps see the following <see href="https://aka.ms/amauthdetails">articles</see> for guidance. </param>
+        public MapsSearchClient(Uri endpoint, TokenCredential credential, string clientId)
+        {
+            Argument.AssertNotNull(credential, nameof(credential));
+
+            endpoint ??= new Uri("https://atlas.microsoft.com");
+            var options = new MapsSearchClientOptions();
+            _clientDiagnostics = new ClientDiagnostics(options);
+            string[] scopes = { "https://atlas.microsoft.com/.default" };
+            _pipeline = HttpPipelineBuilder.Build(options, new BearerTokenAuthenticationPolicy(credential, scopes), new AzureKeyCredentialPolicy(new AzureKeyCredential(clientId), "x-ms-client-id"));
+            RestClient = new SearchRestClient(_clientDiagnostics, _pipeline, endpoint, clientId, options.Version);
+        }
+
+        /// <summary> Initializes a new instance of MapsSearchClient. </summary>
+        /// <param name="credential"> A credential used to authenticate to an Azure Maps Search Service. </param>
+        /// <param name="endpoint"> server parameter. </param>
+        /// <param name="clientId"> Specifies which account is intended for usage in conjunction with the Azure AD security model.  It represents a unique ID for the Azure Maps account and can be retrieved from the Azure Maps management  plane Account API. To use Azure AD security in Azure Maps see the following <see href="https://aka.ms/amauthdetails">articles</see> for guidance. </param>
+        /// <param name="options"> The options for configuring the client. </param>
+        public MapsSearchClient(Uri endpoint, TokenCredential credential, string clientId, MapsSearchClientOptions options)
+        {
+            Argument.AssertNotNull(credential, nameof(credential));
+
+            endpoint ??= new Uri("https://atlas.microsoft.com");
+            options ??= new MapsSearchClientOptions();
+            _clientDiagnostics = new ClientDiagnostics(options);
+            string[] scopes = { "https://atlas.microsoft.com/.default" };
+            _pipeline = HttpPipelineBuilder.Build(options, new BearerTokenAuthenticationPolicy(credential, scopes), new AzureKeyCredentialPolicy(new AzureKeyCredential(clientId), "x-ms-client-id"));
+            RestClient = new SearchRestClient(_clientDiagnostics, _pipeline, endpoint, clientId, options.Version);
+        }
+
+        /// <summary>
+        /// **Get Polygon**
```

Comment:
nit:  This comment is going to render oddly in Intellisense and MS Docs due to the extreme whitespace.    It may be helpful to move the tier information to a `<remarks>` section.   I don't know that the method name here really adds any value.

Leaving one comment, but this seems to apply to the entire class.